src/main/update/extract.ts

```diff
@@ -0,0 +1,57 @@
+import extractZip from "extract-zip";
+import * as utils from "../../utils";
+import { tmpName } from "tmp-promise";
+import { spawn as spawnPromise } from "child-process-promise";
+import fs from "fs";
+
+export async function winExtract(
+  from: string,
+  to: string,
+  win: Electron.BrowserWindow
+) {
+  const tempDir = await tmpName();
+
+  // Unzip ZIP
+  console.log("[player] Start to extract the zip archive", from, "to", tempDir);
+
+  await extractZip(from, {
+    dir: tempDir,
+    onEntry: (_, zipfile) => {
+      const progress = zipfile.entriesRead / zipfile.entryCount;
+      win?.webContents.send("update extract progress", progress);
+    },
+  });
+  win.webContents.send("update extract complete");
+  console.log("[player] The zip archive", from, "has extracted to", tempDir);
+  win.webContents.send("update copying progress");
+  await utils.copyDir(tempDir, to);
```

Comment:
Is there a reason not to extract the target directory directly?

Reply:
Maybe... Nope.
But as I mentioned in this [comment](https://github.com/planetarium/9c-launcher/pull/1727#discussion_r940002597), I tried to use the existing scripts, So I'll try to solve it another PR